crypto-ffi/src/core_crypto/history_observer.rs

```diff
@@ -61,7 +61,7 @@ impl core_crypto::mls::HistoryObserver for ObserverShim {
         };
         if let Err(err) = self
             .0
-            .history_client_created(conversation_id_coerce_maybe_arc(&conversation_id), secret)
+            .history_client_created(Arc::new(ConversationId(conversation_id.as_ref().to_owned())), secret)
```

Comment:
I don't think we need to go through a  reference/to-owned cycle here. We have ownership of the conversation id, so I think this should be sufficient:

```rust
Arc::new(ConversationId(Vec::<u8>::from(conversation_id)))
```